### PR TITLE
Added auth.js to handle login operations, and mutations.js to handle …

### PR DESCRIPTION
**Pull request by RaelNW**
On the client folder under pages i added login and signup files, i have a form for login that pops up as a modal and with chakra styling but to eliminate errors i had to add the auth.js and mutations.js under the utils and sample router code code to main.jsx

### CODE REVIEW

File: client/src/utils/auth.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
+// Importing decode function from 'jwt-decode' library
+import decode from "jwt-decode";
+
+// AuthService class responsible for managing authentication
+class AuthService {
+  // Method to retrieve user profile details from the stored token
```
Comment (briimcfly) [inline client/src/utils/auth.js:6]: @RaelNW do we need the getProfile() method yet? I don't know that we had a profile designed in v1.. I guess this is something we should figure out what we're going to do with profiles.

File: client/src/utils/auth.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  // Method to store the token in local storage upon user login
+  login(idToken) {
+    localStorage.setItem("id_token", idToken); // Sets the token in local storage
+    window.location.assign("/"); // Redirects the user to the home page after login
+  }
+
+  // Method to remove the token from local storage and reload the page upon user logout
+  logout() {
+    localStorage.removeItem("id_token"); // Removes the token from local storage
+    window.location.reload(); // Reloads the current page after logout
```
Comment (briimcfly) [inline client/src/utils/auth.js:47]: @RaelNW we should probably just send the user back to the root here.. 

`window.location.assign('/')` 

thoughts here?

**Review by briimcfly — APPROVED** [review]
Cool! 
I think we're going to move the Signup and Login out of the router eventually, since these will just be popup modals. but for now we can leave them as pages for initial testing then we can refactor later. 

Had a few comments in the PR on the Auth/Mutations pages but I think we can push forward.
Thanks Rael.

File: client/src/utils/auth.js (hunk 1 above)
Comment (RaelNW) [reply client/src/utils/auth.js:6]: We definitely need to discuss on this.

**Pull request merged**